### PR TITLE
[PUBDEV-4956/ITA-185] Java 9 Tests

### PR DESCRIPTION
**Pull request by michal-raska**
Java 9 in docker
It's now possible to specify which Java to use for the stage - default is Java 8
Two new stages - Smoke and JUnits for Java 9

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Thu Feb 02 14:15:44 PST 2017
+#Wed Jan 10 12:12:46 CET 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (mmalohlava) [inline gradle/wrapper/gradle-wrapper.properties]: Please, keep `-all` suffix. it is useful for debugging.

Comment (mmalohlava) [reply gradle/wrapper/gradle-wrapper.properties]: Please make sure that @michalkurka agrees with this change.

Comment (mmalohlava) [reply gradle/wrapper/gradle-wrapper.properties]: one more comment: do not forget to call `./gradlew wrapper` and commit re-generated gradle artefacts.

Comment (michal-raska) [reply gradle/wrapper/gradle-wrapper.properties]: Sure, this is more of an experimental PR

Comment (michal-raska) [reply gradle/wrapper/gradle-wrapper.properties]: > one more comment: do not forget to call ./gradlew wrapper and commit re-generated gradle artefacts.

I think it should be there

Comment (michal-raska) [reply gradle/wrapper/gradle-wrapper.properties]: If I'm correct, than the version 3.3 does not support Java 9 at all. It was the main reason why I tried upgrade in the first place

**Comment by michalkurka** [comment]
@michal-raska great work, as we discussed over the phone - please separate the gradle upgrade to its own PR

File: h2o-core/testMultiNode.sh [hunk 2]
```diff
@@ -76,7 +76,7 @@ else
     COVERAGE=""
 fi
 # Command to invoke test.
-JVM="nice $JAVA_CMD $COVERAGE -Xmx${MAX_MEM} -Xms${MAX_MEM} -ea -cp build/resources/main${SEP}build/resources/test${SEP}build/classes/test${SEP}build/classes/main${SEP}../h2o-genmodel/build/libs/h2o-genmodel.jar${SEP}../lib/*"
+JVM="nice $JAVA_CMD $COVERAGE -Xmx${MAX_MEM} -Xms${MAX_MEM} -ea -cp build/classes/java/test:build/classes/java/main:build/resources/main${SEP}build/resources/test${SEP}build/classes/test${SEP}build/classes/main${SEP}../h2o-genmodel/build/libs/h2o-genmodel.jar${SEP}../lib/*"
```
Comment (mmalohlava) [inline h2o-core/testMultiNode.sh]: btw: the classpath should be created and passed from gradle

File: scripts/jenkins/groovy/defineTestStages.groovy [hunk 3]
```diff
@@ -120,7 +124,11 @@ def call(final pipelineContext) {
       timeoutValue: 10, component: pipelineContext.getBuildConfig().COMPONENT_PY
     ],
     [
-      stageName: 'Java 8 JUnit', target: 'test-junit-jenkins', pythonVersion: '2.7',
+      stageName: 'Java 8 JUnit', target: 'test-junit-jenkins', pythonVersion: '2.7', javaVersion: 8,
+      timeoutValue: 90, component: pipelineContext.getBuildConfig().COMPONENT_JAVA, additionalTestPackages: [pipelineContext.getBuildConfig().COMPONENT_PY]
+    ],
+    [
+      stageName: 'Java 10 JUnit', target: 'test-junit-9-jenkins', pythonVersion: '2.7', javaVersion: 10,
```
Comment (michalkurka) [inline scripts/jenkins/groovy/defineTestStages.groovy]: please rename to test-junit-10-jenkins

**Comment by michal-raska** [comment]
closing in favor of #2454

**Pull request closed without merging**